src/EditorFeatures/Core/Wrapping/BinaryExpression/AbstractBinaryExpressionWrapper.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis.Editor.Wrapping.BinaryExpression
+{
+    internal abstract partial class AbstractBinaryExpressionWrapper<TBinaryExpressionSyntax> : AbstractSyntaxWrapper
+        where TBinaryExpressionSyntax : SyntaxNode
+    {
+        private readonly ISyntaxFactsService _syntaxFacts;
+
+        protected AbstractBinaryExpressionWrapper(ISyntaxFactsService syntaxFacts)
+        {
+            _syntaxFacts = syntaxFacts;
+        }
+
+        protected abstract SyntaxTriviaList GetNewLineBeforeOperatorTrivia(SyntaxTriviaList newLine);
+
+        public sealed override async Task<ICodeActionComputer> TryCreateComputerAsync(
+            Document document, int position, SyntaxNode node, CancellationToken cancellationToken)
+        {
+            if (!(node is TBinaryExpressionSyntax binaryExpr))
+            {
+                return default;
+            }
+
+            if (!IsLogicalExpression(binaryExpr))
+            {
+                return default;
+            }
+
+            // Don't process this binary expression if it's in a parent logical expr.  We'll just
+            // allow our caller to walk up to that and call back into us to handle.  This way, we're
+            // always starting at the topmost logical binary expr.
```

Comment:
❔ Do we need to do anything special here regarding parenthesized expressions?

Reply:
We shouldn't (afaict).  Is there a particular case you were thinking about?  Both for a a binexpr containing a paren'ed sub-expression, and a bin-expr *in* a paren'ed expression, i think we would want to stop at the paren'ed part and not move past it.